### PR TITLE
[WIP] Standard initialization for Incompressible flows using thermodynamic_pressure and the Inc_Temperature_init

### PR DESCRIPTION
**Pull request by Cristopher-Morales**
## Proposed Changes

With this pull request, we aim to initialize the fluid models based on the Ideal gas laws for incompressible flows in a standard way using the THERMODYNAMIC_PRESSURE($P_{op}$) and the INC_TEMPERATURE_INIT ($T_{init}$), the initial density is computed based on the ideal gas law :

$\rho_{init}= P_{op} / R T_{init}$

When the FLUID_MODEL is CONSTANT_DENSITY, the constant density must be given in the config file using the option DENSITY_CONSTANT, similar as it is done when viscosity, thermal conductivity and diffusivity models are chosen as constant.

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [X] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [ ] I used the pre-commit hook to prevent dirty commits and used `pre-commit run --all` to format old commits.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: Common/src/CConfig.cpp [hunk 1]
```diff
@@ -3022,6 +3023,11 @@ void CConfig::SetConfig_Parsing(istream& config_buffer){
             newString.append("DYNAMIC_ANALYSIS is deprecated. Use TIME_DOMAIN instead.\n\n");
           else if (!option_name.compare("SPECIES_USE_STRONG_BC"))
             newString.append("SPECIES_USE_STRONG_BC is deprecated. Use MARKER_SPECIES_STRONG_BC= (marker1, ...) instead.\n\n");
+          else if (!option_name.compare("INC_DENSITY_INIT"))
+            newString.append(
+                "INC_DENSITY_INIT is deprecated. The initial density is computed using the ideal gas law based on the "
+                "THERMODYNAMICS_PRESSURE and INC_TEMPERATURE_INIT. For Using the FLUID_MODEL=CONSTANT_DENSITY, the "
+                "constant density must be given using the following option DENSITY_CONSTANT.\n\n");
```
Comment (pcarruscag) [inline Common/src/CConfig.cpp:3030]: We just released a major version, we are not breaking backward compatibility for an important setting like this.

Comment (bigfooted) [reply Common/src/CConfig.cpp:3030]: Sure, we can wait for a major version to implement a change in config option. But now, sometimes we use thermodynamic pressure and sometimes we use density (to compute thermodynamic pressure) and I think the inconsistency should be fixed at some point, and I think it should be fixed in this way.

Comment (pcarruscag) [reply Common/src/CConfig.cpp:3030]: I'm not saying we cannot fix things, but the option needs to be deprecated more gracefully than this.
As for the inconsistency between density init and pressure init we can use a similar strategy to the compressible solver.
```
/*!
 * \brief Types of initialization option
 */
enum class FREESTREAM_OPTION {
  TEMPERATURE_FS, /*!< \brief Temperature initialization. */
  DENSITY_FS, /*!< \brief Density initalization. */
};
```

**Comment by TobiKattmann** [comment]
I am trying to find some time to think about how to do a bit of option-consolidation on the non-dimensionalization side and initialization is quite a bit linked to that process -> I hope to open a discussion with some more information and proposal soon -> And since this PR is looking to change the INC_DENSITY_INIT option  which is part of that my request would be to maybe spare that option up to the conference (the init change to use thermodynamic pressure is  a little  more independent of that I think via pedros suggestion to mimic what we do for compressible flow atm)

File: Common/include/CConfig.hpp [hunk 2]
```diff
@@ -821,6 +821,7 @@ class CConfig {
   su2double Gamma,      /*!< \brief Ratio of specific heats of the gas. */
   Bulk_Modulus,         /*!< \brief Value of the bulk modulus for incompressible flows. */
   Beta_Factor,          /*!< \brief Value of the epsilon^2 multiplier for Beta for the incompressible preconditioner. */
+  Density_Constant,     /*!< \brief Constant Density for ConstantDensity model (only for incompressible flows). */
```
Comment (TobiKattmann) [inline Common/include/CConfig.hpp:824]: this is a test

**Pull request closed without merging**